File: scala/src/service/OrderService.scala
package service

import model.{Order, OrderStatus}
import repository.OrderRepository

class OrderService {
  private var orderCounter: Long = 0
  private val orders: scala.collection.mutable.ListBuffer[Order] = scala.collection.mutable.ListBuffer()
  private val repository = new OrderRepository()
  
  def createOrder(userId: Long, amount: Double, description: String): Order = {
    orderCounter += 1
    val order = Order(
      id = orderCounter,
      userId = userId,
      amount = amount,
      description = description
    )
    orders.append(order)
    repository.save(order)
    order
  }
  
  def findById(id: Long): Option[Order] = {
    orders.find(_.id == id).orElse(repository.findById(id))
  }
  
  def findByUserId(userId: Long): List[Order] = {
    orders.filter(_.userId == userId).toList
  }
  
  def updateOrderStatus(orderId: Long, newStatus: String): Option[Order] = {
    findById(orderId).map { order =>
      order.status = newStatus
      repository.update(order)
      order
    }
  }
  
  def cancelOrder(orderId: Long): Boolean = {
    updateOrderStatus(orderId, OrderStatus.CANCELLED).isDefined
  }
  
  def getPendingOrders(): List[Order] = {
    orders.filter(_.status == OrderStatus.PENDING).toList
  }
  
  def getAllOrders(): List[Order] = {
    orders.toList
  }
  
  def processOrders(): Int = {
    var processed = 0
    orders.foreach { order =>
      if (order.status == OrderStatus.PENDING) {
        order.status = OrderStatus.PROCESSING
        processed += 1
      }
    }
    processed
  }
  
  def getTotalAmount: Double = {
    orders.map(_.amount).sum
  }
  
  def clearOrders(): Unit = {
    orders.clear()
    orderCounter = 0
  }
}

File: java/src/service/OrderService.java
package com.example.service;

import com.example.exception.ResourceNotFoundException;
import com.example.exception.ValidationException;
import com.example.model.Order;
import com.example.model.OrderStatus;
import com.example.repository.OrderRepository;
import com.example.repository.UserRepository;

import java.util.List;

/**
 * Order business logic layer.
 * Code smell: God class with too many responsibilities
 */
public class OrderService {
    private final OrderRepository orderRepository;
    private final UserRepository userRepository;

    public OrderService() {
        this.orderRepository = OrderRepository.getInstance();
        this.userRepository = UserRepository.getInstance();
    }

    public Order createOrder(Long userId, Double totalAmount, String items) {
        // Code smell: long method with validation scattered
        if (userId == null) {
            throw new ValidationException("userId", "User ID cannot be null");
        }
        if (userId <= 0) {
            throw new ValidationException("userId", "User ID must be positive");
        }
        if (totalAmount == null) {
            throw new ValidationException("totalAmount", "Total amount cannot be null");
        }
        if (totalAmount < 0) {
            throw new ValidationException("totalAmount", "Total amount cannot be negative");
        }
        if (totalAmount > 10000) {
            throw new ValidationException("totalAmount", "Total amount exceeds maximum allowed");
        }

        // Verify user exists
        if (userRepository.findById(userId) == null) {
            throw new ResourceNotFoundException("User", "id", userId);
        }

        if (items == null || items.trim().isEmpty()) {
            throw new ValidationException("items", "Items cannot be empty");
        }

        Order order = new Order(null, userId, totalAmount);
        order.setItems(items);
        return orderRepository.save(order);
    }

    public Order getOrderById(Long id) {
        Order order = orderRepository.findById(id);
        if (order == null) {
            throw new ResourceNotFoundException("Order", "id", id);
        }
        return order;
    }

    public List<Order> getOrdersByUserId(Long userId) {
        if (userId == null) {
            throw new ValidationException("userId", "User ID cannot be null");
        }
        return orderRepository.findByUserId(userId);
    }

    public Order updateOrderStatus(Long orderId, OrderStatus status) {
        if (orderId == null) {
            throw new ValidationException("orderId", "Order ID cannot be null");
        }
        if (status == null) {
            throw new ValidationException("status", "Status cannot be null");
        }

        Order order = orderRepository.findById(orderId);
        if (order == null) {
            throw new ResourceNotFoundException("Order", "id", orderId);
        }

        order.setStatus(status);
        return orderRepository.save(order);
    }

    public Order shipOrder(Long orderId) {
        Order order = getOrderById(orderId);

        if (order.getStatus() != OrderStatus.CONFIRMED) {
            throw new ValidationException("Order must be confirmed before shipping");
        }

        order.setStatus(OrderStatus.SHIPPED);
        order.setShippedAt(java.time.LocalDateTime.now());
        return orderRepository.save(order);
    }

    public List<Order> getAllOrders() {
        return orderRepository.findAll();
    }

    public void cancelOrder(Long orderId) {
        Order order = getOrderById(orderId);

        if (order.getStatus() == OrderStatus.SHIPPED || order.getStatus() == OrderStatus.DELIVERED) {
            throw new ValidationException("Cannot cancel shipped or delivered orders");
        }

        order.setStatus(OrderStatus.CANCELLED);
        orderRepository.save(order);
    }

    public long getOrderCount() {
        return orderRepository.count();
    }
}